Fixed sign in and sign out buttons CSS

andrayantelo opened the pull request:
Also, made the sign in function use sign in with redirect instead of with a popup.

andrayantelo committed:
started firebase integration

andrayantelo committed:
added google sign in and sample data

andrayantelo committed:
starting to incorporate the firebase database

andrayantelo committed:
fixed sign in and sign out button css a little bit

andrayantelo committed:
changed where sign in button is positioned

andrayantelo committed:
moved the sign in button around

andrayantelo committed:
Fixed the sign in and sign out button CSS

johntyree commented on .firebaserc:
```diff
@@ -0,0 +1,6 @@
+    "default": "check-it-14d35",
+    "checkit": "check-it-14d35"
+  }
+}
```
File should end with a newline.

johntyree replied:
missing newline

johntyree commented on database.rules.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "rules": {
+    ".read": "auth != null",
```
This is OK for now. You'll want to further limit this later so that users can only read / write their own data.

andrayantelo committed:
Merge branch 'master' into signInButton

johntyree commented on public/calendar.css:
```diff
@@ -6,6 +6,63 @@ body{
+}
+
+#user-container {
+  display: flex;
```
My understanding is that this isn't cross platform yet and that you have to use the -moz and -webkit versions in addition to this. Maybe look into that?

johntyree commented on database.rules.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "rules": {
+    ".read": "auth != null",
```
You'll want to limit this such that users can only read/write their own data eventually. (open an issue?)

andrayantelo replied:
Opened issue #107

johntyree commented on public/calendar_reboot.js:
```diff
@@ -94,10 +109,81 @@ function CheckIt() {
            console.log("There is no current active calendar");
            this.showBuildMenu();
-       }.bind(this));
+       }.bind(this));   
```
There is whitespace (ws) at the end of this line

andrayantelo committed:
fix white space errors

andrayantelo committed:
Merge branch 'signInButton' of https://github.com/andrayantelo/calendar-reboot into signInButton

andrayantelo committed:
added cross browser compatibility for display:flex

andrayantelo committed:
deleted a variable I never use

johntyree commented on public/calendar_reboot.js:
```diff
@@ -94,10 +108,81 @@ function CheckIt() {
+      }
+    else { // User is signed out!
+        // Remove the user from the store so that we can't access their information.
+        this.store.user = null;
```
What about the active calendar and the list of calendars in the drop down?

johntyree replied:
If the user has signed out, they should no longer be available.

johntyree commented on public/index.html:
```diff
@@ -90,18 +111,33 @@
+            <div id="user-container">
+                <div hidden id="user-pic"></div>
+                <div hidden id="user-name"></div>
+                <button class="btn btn-primary navbar-btn " hidden id="sign-out"><!-- mdl-js-button mdl-js-ripple-effect mdl-color-text--white"-->
```
What is commented out here? Should it be uncommented or should it be removed?

andrayantelo replied:
Should be removed.

andrayantelo committed:
removed commented out lines

andrayantelo committed:
Fixed filling dropdown and displaying active cal
Made methods fillDropdown and displayActiveCalendar
to be run when the user signs in instead of on page load.

andrayantelo committed:
Hide/Show build form on sign in and sign out
Also, have a 'get started'  blurb that is visible
when user is not signed in, and invisible when signed
in.

johntyree commented on public/index.html:
```diff
@@ -90,27 +114,45 @@
+                <button class="btn btn-primary navbar-btn " hidden id="sign-out">
+                  Sign-out
+                </button>
+                <button class="btn btn-primary navbar-btn " hidden id="sign-in"> 
```
Why does the sign-in button start out hidden?

johntyree replied:
Apparently, `firebase.auth` always chooses one of the two buttons to unhide.

What about the other buttons?

andrayantelo committed:
added hidden attribute to build form and get started blurb
